src/domain/application/application.entity.ts

```diff
@@ -0,0 +1,78 @@
+import { MID_TEXT_LENGTH } from '@constants';
+import { Challenge } from '@domain/challenge/challenge.entity';
+import { Ecoverse } from '@domain/ecoverse/ecoverse.entity';
+import { NVP } from '@domain/nvp/nvp.entity';
+import { Opportunity } from '@domain/opportunity/opportunity.entity';
+import { User } from '@domain/user/user.entity';
+import { Field, ID, ObjectType, registerEnumType } from '@nestjs/graphql';
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { Question } from '@domain/application/application.dto';
+
+export enum ApplicationStatus {
+  new,
+  approved,
+  rejected,
+}
+
+registerEnumType(ApplicationStatus, {
+  name: 'ApplicationStatus',
+});
+
+@Entity()
+@ObjectType()
+export class Application extends BaseEntity {
+  @Field(() => ID)
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Field(() => ApplicationStatus, { nullable: false })
+  @Column()
+  status!: ApplicationStatus;
+
+  @Field(() => String, { nullable: true })
+  @Column({ length: MID_TEXT_LENGTH, nullable: true })
+  reason?: string;
+
+  @Field(() => User)
+  @ManyToOne(
+    () => User,
+    user => user.focalPoints,
+    { eager: true, cascade: true, onDelete: 'CASCADE' }
+  )
+  user!: User;
+
+  @Field(() => [Question])
+  @ManyToMany(
```

Comment:
Questions are not going to be shared i.e. each question entity is specific to a particular applicaiton.

So this can be oneToMany. Ditto for all ther manyToMany as far as I can tell i.e. applications are also not re-used across entities.

Reply:
The Questions are using the Generic entity `nvp` so it is not possible to use OneToMany. If we specify OneToMany/ManyToOne the nvp entity will have ApplicationId as foreign key, and every future entities having such relation will amend that table. To resolve that we are defining the ralation like ManyToMany, but we are going to use it like `1:n`